include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -75,125 +92,269 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         int m_nknots;
 
     public:
+        /// @brief The type of the discrete dimension representing the B-splines.
         using discrete_dimension_type = NonUniformBSplines;
 
+        /// @brief The type of a DiscreteDomain whose elements identify the B-splines.
         using discrete_domain_type = DiscreteDomain<DDim>;
 
+        /// @brief The type of a DiscreteElement identifying a B-spline.
         using discrete_element_type = DiscreteElement<DDim>;
 
+        /// @brief The type of a DiscreteVector representing an "index displacement" between two B-splines.
         using discrete_vector_type = DiscreteVector<DDim>;
 
         Impl() = default;
 
-        template <class OriginMemorySpace>
-        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
-            : m_domain(impl.m_domain)
-            , m_nknots(impl.m_nknots)
-        {
-        }
-
-        /// @brief Construct a `Impl` using a brace-list, i.e. `Impl bsplines({0., 1.})`
+        /** @brief Constructs an Impl using a brace-list, i.e. `Impl bsplines({0., 1.})`
+         *
+         * The brace-list is usually the iterable list of knots. However, only the first and last
+         * indexes of the elements of the lists are effectively used by the constructor to build the Impl.
+         */
```

Comment:
I think that `breaks` represents the break points not the knots.

> However, only the first and last indexes of the elements of the lists are effectively used by the constructor

This is not true. `breaks.begin()` and `breaks.end()` are not indexes they are iterators. `breaks.end()` is not the last element of the list, it is the one after (the stop condition for the iterator). This constructor calls another constructor below. It uses all the elements